src/lib/util.js

```diff
@@ -233,6 +233,29 @@ const fetchParticipantInfo = async (source, destination, cache) => {
   return { payer, payee }
 }
 
+const getParticipantEndpoint = async ({ fspId, db, loggerFn, endpointType, proxyClient = null }) => {
+  if (!fspId || !db || !loggerFn || !endpointType) {
+    throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.INTERNAL_SERVER_ERROR, 'Missing required arguments for \'getParticipantEndpoint\'')
+  }
+
+  let endpoint = await db.getParticipantEndpoint(fspId, endpointType)
```

Comment:
I am a bit surprised to see that this service calls the DB directly. `ml-api-adapter` gets the endpoint through a REST API call, which it does by using a function from central-services-shared. The PR there already adds the proxying capabilities. If there is future development related to proxy, using the API and central-services-shared will save us the effort to implement in multiple places.

Reply:
@kalinkrustev Agreed. This was discussed with @PaulGregoryBaker. It's a technical debt that needs to be paid.
Also note, the implementation also caches calls to the db.